File: back/src/main/java/com/codejourney/pimber/entity/Party.java
package com.codejourney.pimber.entity;

import jakarta.persistence.*;
import lombok.Data;

import java.time.LocalDateTime;
import java.util.Set;
import java.util.UUID;

@Entity
@Data
public class Party {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    private String name;

    private int count_people;

    private int max_people;

    private LocalDateTime start_timestamp;

    private LocalDateTime end_timestamp;

    private LocalDateTime publication;

    private float price;

    private boolean bring_something;

    @ManyToOne(fetch = FetchType.LAZY, cascade = CascadeType.ALL)
    private Type type;

    @ManyToOne(fetch = FetchType.LAZY, cascade = CascadeType.ALL)
    private Address address;

    @ManyToMany(fetch = FetchType.LAZY)
    private Set<PartyVideoGame> videoGames;

    @ManyToMany(fetch = FetchType.LAZY)
    private Set<BoardGame> boardGames;

    @OneToMany(fetch = FetchType.LAZY, cascade = CascadeType.ALL)
    private Set<Photo> photos;

    @ManyToOne(fetch = FetchType.LAZY, cascade = CascadeType.ALL)
    private User creator;

    @ManyToMany(fetch = FetchType.LAZY)
    private Set<User> invitedUsers;
}

File: front/src/types/party.ts
import { Address } from "./address"
import { BoardGame } from "./boardGame"
import { Photo } from "./photo"
import { Type } from "./type"
import { User } from "./user"
import { VideoGame } from "./videoGame"

export type Party = {
  id: string
  name: string
  count_people: number
  max_people: number
  start_timestamp: Date
  end_timestamp: Date
  publication: Date
  price: number
  bring_something: boolean
  type?: Type
  address?: Address
  videoGames?: VideoGame[]
  boardGames?: BoardGame[]
  photos: Photo[]
  creator?: User
  invitedUsers: User[]
}

export type PartyForm = {
  name: string
  count_people: number
  max_people: number
  start_timestamp: Date
  end_timestamp: Date
  publication: Date
  price: number
  bring_something: boolean
  type: { id: string }
  address: { id: string }
  photos: Photo[]
  creator: { id: string }
}
